### PR TITLE
feat: add meeting preview modal

### LINKED ISSUE #3431: Add details modal for timeline meet stubs
design: https://www.figma.com/file/QsyWtRJW9eoLB7eZkVY8y7/Timeline?node-id=843%3A47263&mode=dev

### PR DESCRIPTION
**Pull request by acalinica**
## Proposed changes

closes #3431 

## Changes

What types of changes does your code introduce?  _Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe below)

### CODE REVIEW

File: packages/apps/spaces/app/organizations/[id]/components/Timeline/events/meeting/MeetingPreviewModal.tsx [hunk 1]
```diff
@@ -0,0 +1,339 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+import { convert } from 'html-to-text';
+import { useForm } from 'react-inverted-form';
+import { utcToZonedTime, format } from 'date-fns-tz';
```
Comment (Kasia9090) [inline packages/apps/spaces/app/organizations/[id]/components/Timeline/events/meeting/MeetingPreviewModal.tsx]: Move those to DateUtils class, it's to keep those fuctions after fasade to ensure easiness on changing library if we'll ever want to do that for some reason

File: packages/apps/spaces/app/organizations/[id]/components/Timeline/events/meeting/MeetingPreviewModal.tsx [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+
+  const owner = (() => {
+    if (event?.createdBy?.[0]?.__typename === 'ContactParticipant') {
+      const participant = event?.createdBy?.[0]?.contactParticipant;
+      return participant.emails?.[0]?.email ?? participant?.firstName;
+    }
+    return '';
+  })();
+
+  const participants = event?.attendedBy
```
Comment (Kasia9090) [inline packages/apps/spaces/app/organizations/[id]/components/Timeline/events/meeting/MeetingPreviewModal.tsx:84]: This does not include all possible types of participants. Please use helper function from timeline utils to ensure all possible types of participants are handled in consistent way

**Review by Kasia9090 — CHANGES_REQUESTED** [review]

File: packages/apps/spaces/app/organizations/[id]/components/Timeline/events/utils.ts [hunk 3]
```diff
@@ -7,40 +7,43 @@ import {
 } from '@graphql/types';
 import { getName } from '@spaces/utils/getParticipantsName';
 
-
-export const getParticipantName = (
+export const getParticipantEmailOrName = (
```
Comment (Kasia9090) [inline packages/apps/spaces/app/organizations/[id]/components/Timeline/events/utils.ts]: Do we prefer email over name in stubs as well? I think we need two functions one that prefers name (stubs, other types of interaction event participants) and one that prefer email over name for meeting modal view and possibly other events in the future

File: packages/apps/spaces/app/organizations/[id]/components/Timeline/events/meeting/MeetingPreviewModal.tsx [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+      event?.endedAt,
+      DateTimeUtils.timeWithGMT,
+      creatorTimeZoneCode,
+    );
+
+    return `${start} - ${end}`;
+  })();
+
+  const owner = (() => {
+    if (event?.createdBy?.[0]?.__typename === 'ContactParticipant') {
```
Comment (Kasia9090) [inline packages/apps/spaces/app/organizations/[id]/components/Timeline/events/meeting/MeetingPreviewModal.tsx]: Event participant can be also user or organization

**Review by Kasia9090 — CHANGES_REQUESTED** [review]

File: packages/apps/spaces/app/organizations/[id]/components/Timeline/events/utils.ts (hunk 3 above)
Comment (acalinica) [reply packages/apps/spaces/app/organizations/[id]/components/Timeline/events/utils.ts]: Yes, I've just checked before pushing that for Meetings we prefer emails first and names after. For other cases what can I say, I initially kept the code co-located only to meet stuff, now it became more generic which I would've not.

Comment (acalinica) [reply packages/apps/spaces/app/organizations/[id]/components/Timeline/events/utils.ts]: @steynviljoen67 can confirm this to you.

Comment (steynviljoen67) [reply packages/apps/spaces/app/organizations/[id]/components/Timeline/events/utils.ts]: In stubs, prefer name first. If no name is available, show email.

**Review by Kasia9090 — APPROVED** [review]

**Pull request merged**